modules/hdf4_handler/hdfdesc.cc

```diff
@@ -5867,11 +5909,200 @@ bool obtain_eos2_gd_ll_info(const string & fname, const string & grid_name, int3
     if (GCTP_CEA == projcode || GCTP_GEO == projcode)
         oned_ll = true;
 
+    eg_info.projcode = projcode;
+    eg_info.zone = zone;
+    eg_info.sphere = sphere;
+    for (int i = 0; i<2;i++) {
+        eg_info.upleft[i]   = upleft[i];
+        eg_info.lowright[i] = lowright[i];
+    }
+    for (int i = 0; i<16;i++)
+        eg_info.params[i] = params[i];
+  
     GDdetach(gridid);
     GDclose(gfid);
 
     return true;
 }
+void add_dummy_grid_cv(D4Group *d4_grp, const eos2_grid_info_t &eg_info) {
+
+    string dummy_proj_cf_name = "eos5_cf_projection";
+    auto dummy_proj_cf_unique = make_unique<HDFEOS2GeoCFProj>(dummy_proj_cf_name, dummy_proj_cf_name);
+    HDFEOS2GeoCFProj *dummy_proj_cf = dummy_proj_cf_unique.release();
```

Comment:
Instead of calling release here, you could call get(). That gets the pointer, so you can easily pass it to the various functions but keeps dummy_proj_cf_unique in control of the pointer in case an exception is thrown. 

Alternatively, you could each of the calls like:
         add_var_dap4_attr(dummy_proj_cf_unique.get(), "longitude_of_central_meridian", attr_float64_c, "0.0");

Then down below...

Reply:
I will merge the code for now. We can discuss this later.